lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -671,12 +685,23 @@ def _push_alerts_to_relation_databag(self, relation: Relation) -> None:
 
         alert_rules_as_dict = alert_rules.as_dict()
 
+        if extra_alerts_list:
+            if alert_rules_as_dict:
+                alert_rules_as_dict["groups"].extend(extra_alerts_list)
+            else:
+                alert_rules_as_dict["groups"] = extra_alerts_list
+
         if alert_rules_as_dict:
             relation.data[self._charm.app]["alert_rules"] = json.dumps(alert_rules_as_dict)
 
-    def reload_alerts(self) -> None:
+    def reload_alerts(self, *, extra_alerts: Optional[dict] = None) -> None:
         """Reload alert rules from disk and push to relation data."""
-        self._push_alerts_to_all_relation_databags(None)
+        extra_alerts_list = []
+        if extra_alerts:
```

Comment:
Based on the function name/doc it seems like a reload should occur with or without `extra_alerts` present.

If that is not the case then we could simplify indentation by using a guard:
```python
if not extra_alerts:
    return
```

Reply:
in this case there are interal functions that re-read from disk. Don't think we should exit.
That also will break back-compatibility

Reply:
I don't follow. Afaict, current code is equivalent to having
```python
if not extra_alerts:
    return
```
at the top.